src/allotropy/parsers/appbio_quantstudio_designandanalysis/appbio_quantstudio_designandanalysis_calculated_documents.py

```diff
@@ -243,55 +244,58 @@ def build_delta_ct_mean(
 
 @cache
 def build_delta_ct_sd(
-    view_data: ViewData[WellItem], sample: str, target: str, r_target: str
+    view_data: ViewData[WellItem], sample: str, target: str, r_target: str | None
 ) -> CalculatedDocument | None:
     well_items = view_data.get_leaf_item(sample, target)
     if (delta_ct_sd := well_items[0].result.delta_ct_sd) is None:
         return None
 
-    ct_sd_ref = build_ct_sd(view_data, sample, target)
-    if ct_sd_ref is None:
-        return None
+    data_sources = []
 
-    r_ct_sd_ref = build_ct_sd(view_data, sample, r_target)
-    if r_ct_sd_ref is None:
+    if ct_sd_ref := build_ct_sd(view_data, sample, target):
+        data_sources.append(DataSource(feature="ct sd", reference=ct_sd_ref))
+    else:
         return None
 
+    if r_target is not None:
```

Comment:
If r_target is None, I think you want to return None here, right?

If so, suggest to move `if r_target is None -> return None` to the top.

Reply:
is the same situation as above, if r_target is none I just have to ignore the invocation of build_ct_sd with r_target
because data_sources contains a value that I can't ignore.